src/logic/safe/api/fetchSafeTxGasEstimation.ts

```diff
@@ -11,8 +16,18 @@ type FetchSafeTxGasEstimationProps = {
 export const fetchSafeTxGasEstimation = async ({
   safeAddress,
   ...body
-}: FetchSafeTxGasEstimationProps): Promise<string> => {
-  return postSafeGasEstimation(GATEWAY_URL, _getChainId(), checksumAddress(safeAddress), body).then(
-    ({ safeTxGas }) => safeTxGas,
-  )
+}: FetchSafeTxGasEstimationProps): Promise<SafeTransactionEstimation> => {
+  return postSafeGasEstimation(GATEWAY_URL, _getChainId(), checksumAddress(safeAddress), body)
+}
+
+export const getRecommendedNonce = async (safeAddress: string): Promise<number> => {
+  const { recommendedNonce } = await fetchSafeTxGasEstimation({
+    safeAddress,
+    value: '0',
+    operation: Operation.CALL,
+    // Workaround: use a cancellation transaction to fetch only the recommendedNonce
```

Comment:
In future, we should make just one request – to fetch the actual safeTxGas _and_ the nonce. They are displayed on the same Review screen.